bundles/test.rst

```diff
@@ -0,0 +1,94 @@
+TestBundle
+==========
+
+Writing automatic tests for a Sulu project will be similar to tests in a usual
+Symfony project. Therefore you should have a look at their `test documentation`_.
+
+
+Sulu's Kernel Context
+---------------------
+
+Sulu adds an additional layer to the service container of Symfony. This is
+the kernel context. It separates the *admin* area from the *website* area.
+
+When you start writing integration or functional tests you have to keep in mind
+that some services are only available in either context.
+
+Integration Tests
+-----------------
+
+In integration tests you may have depencencies on other services. If a service
+is only available in the website context you will get issues, because the
+kernel is in admin context per default.
+
+Therefore Sulu provides an extended `KernelTestCase`_ to specify the kernel
+context.
+
+.. code-block:: php
+    
+    // tests/Integration/Service/NewsletterGeneratorTest.php
+    namespace App\Tests\Integration\Service;
+    
+    use Sulu\Bundle\TestBundle\Testing\KernelTestCase;
+    use Sulu\Component\HttpKernel\SuluKernel;
+    
+    class NewsletterGeneratorTest extends KernelTestCase
+    {
+        public function testSomething()
+        {
+            self::bootKernel([
+                'sulu.context' => SuluKernel::CONTEXT_WEBSITE
+            ]);
+
+            // ...
+        }
+    }
+
+This will boot the kernel in the website context so that you have access to
+the services living in it.
+
+Functional Tests
+----------------
+
+In functional tests you test your application from a higher level. Instead of
+single methods or algorithms you run the functions from the same level as the
+user.
+
+Similar to integration tests you also have to boot the kernel in the website
+context if you want to test controllers and services living in that context.
+
+Use Sulu's `SuluTestCase`_ to create a client in the website context and define
+your expectations to the called action.
+
+.. code-block:: php
+    
+    // tests/Functional/Controller/Website/RegistrationControllerTest.php
+    namespace App\Tests\Functional\Controller\Website;
+    
+    use Sulu\Bundle\TestBundle\Testing\SuluTestCase;
+    use Sulu\Component\HttpKernel\SuluKernel;
+    
+    class RegistrationControllerTest extends SuluTestCase
+    {
+        public function testIndexAction(): void
+        {
+            $client = static::createClient([
+                'sulu.context' => SuluKernel::CONTEXT_WEBSITE,
+            ]);
```

Comment:
there is a short helper function for this `static::createWebsiteClient()`: https://github.com/sulu/sulu/blob/0b073109244bfa99def7fb63afdc0c62e9b66980/src/Sulu/Bundle/TestBundle/Testing/CreateClientTrait.php#L50

Think it would also be create that we document the other 2 methods in there and that there is also helper function to get a user without manually creating one: `getTestUser` function.

Which is active only on test environment via: https://github.com/sulu/skeleton/blob/c8a9dcc4e03e627924fb28495e3fe0ac98906828/config/packages/security.yaml#L116-L117

Reply:
I have updated the PR so that the helper function is shown.

Regarding your second item, I have documented the `getTestUser` function in combination with Symfony's new [`$client->loginUser($testUser)`](https://symfony.com/doc/current/testing.html#logging-in-users-authentication) method. Because I think it is simpler as the `createAuthenticated{Website}Client` methods.

Additionally I have added a further paragraph about the database purging helper.